src/power.rs

```diff
@@ -28,6 +39,76 @@ impl Power {
         Self { rb: pwr }
     }
 
+    pub fn get_standby_flag(&mut self) -> bool {
+        self.rb.sr1.read().sbf().bit_is_set()
+    }
+
+    pub fn get_wakeup_flag<L: Into<WakeUp>>(&self, lane: L) -> bool {
+        match lane.into() {
+            WakeUp::Line1 => self.rb.sr1.read().wuf1().bit_is_set(),
+            WakeUp::Line2 => self.rb.sr1.read().wuf2().bit_is_set(),
+            WakeUp::Line4 => self.rb.sr1.read().wuf4().bit_is_set(),
+            WakeUp::Line5 => self.rb.sr1.read().wuf5().bit_is_set(),
+            WakeUp::Line6 => self.rb.sr1.read().wuf6().bit_is_set(),
+            _ => false,
+        }
+    }
+
+    pub fn clear_wakeup_flag<L: Into<WakeUp>>(&mut self, lane: L) {
+        match lane.into() {
+            WakeUp::Line1 => self.rb.scr.write(|w| w.cwuf1().set_bit()),
+            WakeUp::Line2 => self.rb.scr.write(|w| w.cwuf2().set_bit()),
+            WakeUp::Line4 => self.rb.scr.write(|w| w.cwuf4().set_bit()),
+            WakeUp::Line5 => self.rb.scr.write(|w| w.cwuf5().set_bit()),
+            WakeUp::Line6 => self.rb.scr.write(|w| w.cwuf6().set_bit()),
+            _ => {}
+        }
+    }
+
+    pub fn clear_standby_flag(&mut self) {
+        self.rb.scr.write(|w| w.csbf().set_bit());
+    }
+
+    pub fn enable_wakeup_lane<L: Into<WakeUp>>(&mut self, lane: L, edge: SignalEdge) {
```

Comment:
I guess all these functions consume a GPIO pin without a way to get it back?

Reply:
Nope, `From` trait implemented for `&$PIN`